src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -254,6 +256,36 @@ private void ApplyChangeSetForceExecute(ChangeSetData changeSet)
             }
         }
 
+        private void ReActivateGraphNodesInCycle(List<AssociativeNode> nodeList)
+        {
+            if (nodeList == null || !nodeList.Any()) return;
+
+            var assocGraph = core.DSExecutable.instrStreamList[0].dependencyGraph;
+            var graphNodes = assocGraph.GetGraphNodesAtScope(Constants.kInvalidIndex, Constants.kInvalidIndex);
+
+            foreach (var node in nodeList)
+            {
+                var bNode = node as BinaryExpressionNode;
+
+                var identifier = bNode?.LeftNode as IdentifierNode;
+                if (identifier == null) continue;
+
+                GraphNode rootNode = null;
+                foreach(var gNode in graphNodes)
```

Comment:
is this all graphNodes?

Reply:
No, graphNodes are only those that are in global scope (class index and function index = -1 or invalid index).

Reply:
I can't think of a case where we can have a cycle in a single code block node scope or function definition scope as we disallow variable redefinitions in code block nodes 🤔